### PR TITLE
save_file method receives string object

### PR DESCRIPTION
**Pull request by kojix2**
https://github.com/red-data-tools/red-chainer/issues/44
I think this is almost as same as the original Chainer.
Please fix it to Red Chainer style.

### CODE REVIEW

File: lib/chainer/serializers/marshal.rb [hunk 1]
```diff
@@ -4,6 +4,12 @@ class MarshalSerializer < Chainer::Serializer
       attr_accessor :target, :path
 
       def self.save_file(filename, obj)
+        if filename.instance_of?(String)
+          File.open(filename, 'wb') do |f|
+            self.save_file(f, obj)
+          end
+          return
+        end
```
Comment (hatappi) [inline lib/chainer/serializers/marshal.rb]: I thought it strange that original implementation.
`File` object is passed to `filename`.


What do you think is natural thought as follows?

```
s = self.new
s.save(obj)
File.open(filename, 'wb') do |f|
   Marshal.dump(s.target, f)
end
```

**Comment by hatappi** [comment]
@kojix2 
I comment this p-r.
Please check comment 👍

File: lib/chainer/serializers/marshal.rb (hunk 1 above)
Comment (kojix2) [reply lib/chainer/serializers/marshal.rb]: Well...
When calling this save_file function from elsewhere,  snapshot for example, the file object will be passed.

Comment (hatappi) [reply lib/chainer/serializers/marshal.rb]: I am planning to respond like this.
https://github.com/red-data-tools/red-chainer/pull/47/files

Comment (kojix2) [reply lib/chainer/serializers/marshal.rb]: I’ll leave it up to you.

**Pull request closed without merging**